pkg/infra/kubernetes/exec_unit.go

```diff
@@ -173,12 +173,27 @@ func (unit *HelmExecUnit) transformDeployment() ([]HelmChartValue, error) {
 	if err != nil {
 		return nil, err
 	}
-	values = append(values, HelmChartValue{
-		ExecUnitName: unit.Name,
-		Kind:         deployment.Kind,
-		Type:         string(ImageTransformation),
-		Key:          value,
-	})
+	values = append(values,
+		HelmChartValue{
+			ExecUnitName: unit.Name,
+			Kind:         deployment.Kind,
+			Type:         string(ImageTransformation),
+			Key:          value,
+		},
+		HelmChartValue{
+			ExecUnitName: unit.Name,
+			Kind:         deployment.Kind,
+			Type:         string(ExecUnitNetworkPlacement),
+			Key:          fmt.Sprintf("{{ .Values.%sNetworkPlacement }}", unit.Name),
+		},
+		HelmChartValue{
+			ExecUnitName: unit.Name,
+			Kind:         deployment.Kind,
+			Type:         string(ExecUnitNodeGroup),
+			Key:          fmt.Sprintf("{{ .Values.%sNodeGroup }}", unit.Name),
+		},
+	)
```

Comment:
this value may not always exist. If the user doesnt specify any constrainst on an exec unit we dont need to care what node it gets launched on